rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java

```diff
@@ -108,4 +110,17 @@
         }
         return rs;
     }
+
+    private static boolean requiresUndoFunction() {
+
+        if (!RootContext.inGlobalTransaction() && !RootContext.requireGlobalLock()) {
+            return false;
+        }
+
+        if (RootContext.inGlobalTransaction() && StringUtils.equals(BranchType.TCC.name(), RootContext.getBranchType())) {
```

Comment:
Add BranchType.SAGA. Bind Saga type will commit at another pr : https://github.com/seata/seata/pull/2551

Reply:
OK,thank you for notification, branchType SAGA was now added in my judgement conditions.